api/provider/task.go

```diff
@@ -3,8 +3,9 @@ package provider
 import "github.com/quintilesims/layer0/common/models"
 
 type TaskProvider interface {
-	Create(req models.CreateTaskRequest) (*models.Task, error)
-	Delete(taskID string) error
-	List() ([]models.TaskSummary, error)
+	// todo: all creates should return a string
+	Create(req models.CreateTaskRequest) (string, error)
 	Read(taskID string) (*models.Task, error)
+	List() ([]models.TaskSummary, error)
+	Delete(taskID string) error
```

Comment:
[nit] I think we decided to make these alphabetical, not CRUD-ordered (made the same comment on Josh's WIP PR)

Reply:
:+1: